src/item.cpp

```diff
@@ -2666,67 +2665,69 @@ void item::armor_info( std::vector<iteminfo> &info, const iteminfo_query *parts,
                         }
                     }
                 }
-
-                // Handle things that use both sides to avoid showing L Arm R Arm etc when not needed
+                // Handle things that use both sides to avoid showing L. Arm R. Arm etc when both are the same
                 if( !t->sided ) {
-                    if( covers( bodypart_id( "foot_l" ) ) && covers( bodypart_id( "foot_r" ) )
-                        && encumb_data.at( bodypart_str_id( "foot_r" ) ).first == encumb_data.at(
-                            bodypart_str_id( "foot_l" ) ).first ) {
-                        encumb_data.erase( encumb_data.find( bodypart_str_id( "foot_l" ) ) );
-                        encumb_data[bodypart_str_id( "Feet" )] = encumb_data.at( bodypart_str_id( "foot_r" ) );
-                        encumb_data[bodypart_str_id( "Feet" )].second.to_display =
-                            encumb_data[bodypart_str_id( "Feet" )].second.name_as_heading_multiple;
-                        encumb_data.erase( encumb_data.find( bodypart_str_id( "foot_r" ) ) );
+                    bodypart_str_id right_foot = bodypart_id( "foot_l" )->opposite_part;
+                    bodypart_str_id right_leg = bodypart_id( "leg_l" )->opposite_part;
+                    bodypart_str_id right_arm = bodypart_id( "arm_l" )->opposite_part;
+                    bodypart_str_id right_hand = bodypart_id( "hand_l" )->opposite_part;
+                    bodypart_str_id left_foot = bodypart_id( "foot_r" )->opposite_part;
+                    bodypart_str_id left_leg = bodypart_id( "leg_r" )->opposite_part;
+                    bodypart_str_id left_arm = bodypart_id( "arm_r" )->opposite_part;
+                    bodypart_str_id left_hand = bodypart_id( "hand_r" )->opposite_part;
+                    bodypart_str_id feet( "Feet" );
+                    bodypart_str_id arms( "Arms" );
+                    bodypart_str_id legs( "Legs" );
+                    bodypart_str_id hands( "Hands" );
+
+                    if( covers( left_foot ) && covers( right_foot )
+                        && piece_data.at( left_foot ).first == piece_data.at( right_foot ).first ) {
+                        piece_data.erase( piece_data.find( left_foot ) );
+                        piece_data[feet] = piece_data.at( right_foot );
+                        piece_data[feet].second.to_display = piece_data[feet].second.multiple_name;
+                        piece_data.erase( piece_data.find( right_foot ) );
                     }
```

Comment:
This is better, but you shouldn't need to hardcode all the body parts like this.

You should be able to loop over bodyparts `bp` and do something like

```c++
bodypart_str_id opposite = bp->opposite_part;
if( covers( bp ) && covers( opposite )
    && piece_data.at( bp ).first == piece_data.at( opposite ).first ) {
    piece_data.erase( bp );
    piece_data[bp].second.to_display = piece_data[bp].second.multiple_name;
    piece_data.erase( opposite );
}
```

Reply:
Thanks for this idea, I didn't think about iterating through them due to lack of "new style" array, but I can simply convert them over. The only problem with this is that now we have no real way of converting "L. Arm" into "Arms" to display to the player.

Reply:
I imagine you've figured it out, but the `piece_data.erase( bp );` I wrote above shouldn't be there.